### PR TITLE
test: add comprehensive unit tests for shap/utils/image.py

### PR DESCRIPTION
**Pull request by aryanyk**
## Description
This PR adds comprehensive unit tests for `shap/utils/image.py` to increase code coverage from 0% to ~90%.

## Changes
- ✅ Added 18 test cases covering all 8 public functions
- ✅ Tests cover normal usage, edge cases, and error handling
- ✅ All tests use proper mocking to avoid external dependencies
- ✅ Tests are deterministic and cross-platform compatible

## Coverage Impact
- **Before**: 0% coverage for `shap/utils/image.py`
- **After**: ~90% coverage (verified with pytest-cov)

## Functions Tested
- `is_empty()` - 3 tests
- `make_dir()` - 3 tests  
- `add_sample_images()` - 1 test
- `load_image()` - 2 tests
- `check_valid_image()` - 1 parametrized test (8 cases)
- `save_image()` - 2 tests
- `resize_image()` - 4 tests
- `display_grid_plot()` - 2 tests

## Checklist
- [x] Tests pass locally
- [x] No linting errors
- [x] Coverage improved significantly
- [x] Tests are deterministic
- [x] Follows pytest best practices

## Related Issues
Addresses #3801 

## Screenshots 
image

### CODE REVIEW

File: tests/utils/test_image.py [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+
+    calls = []
+
+    def fake_imagenet50():
+        return images, labels
+
+    def fake_save_image(arr, path):
+        calls.append((arr.copy(), path))
+
+    monkeypatch.setattr(image_utils.shap.datasets, "imagenet50", fake_imagenet50)
```
Comment (CloseChoice) [inline tests/utils/test_image.py:84]: can you pls load one of the images in tests/plots/baseline?

**Review by CloseChoice — CHANGES_REQUESTED** [review]
Can we pls get rid of the mocking in here where possible?

**Comment by aryanyk** [comment]
Hi @CloseChoice,

Thanks for the feedback. I’ve updated the tests to remove mocking where possible and now use baseline images from `tests/plots/baseline`. Would appreciate your review.
